client/src/pages/game/Game.js

```diff
@@ -1,20 +1,75 @@
-import React from "react";
+import React, { useState, useEffect, useContext } from "react";
+import Button from "@material-ui/core/Button";
 import Container from "@material-ui/core/Container";
 import Grid from "@material-ui/core/Grid";
 
+import { AppContext } from "../../App";
+import { useGameState } from "../../socket_io/GameIO";
+import GamePrompt from "./GamePrompt";
 import GameBoard from "./GameBoard";
 import Chat from "../chat/Chat";
+
+import GameIO from "../../socket_io/GameIO";
 import "./Game.css";
 
-function Game() {
+function Game(props) {
+  const { gameIO } = useContext(AppContext);
+  const [matchId, setMatchId] = useState("");
+  const [player, setPlayer] = useState(null);
+  const [gameState, setGameState] = useState(props.location.state ? props.location.state.gameState : null);
+
+  useEffect(() => {
+    // set game data from game lobby data
+    const matchId = props.location.state
+      ? props.location.state.matchId
+      : null; 
+    const user = props.location.state
+      ? props.location.state.user
+      : null; 
+    const role = props.location.state ? props.location.state.role : "Spymaster"; // dummy data
+    const player = { user, role };
+    
+    if (matchId && user) {
+      setMatchId(matchId);
+      setPlayer(player);
+      setGameState(gameState);
```

Comment:
We don't need this setGameState here anymore. We can remove this line.

Reply:
👌